internal/cri/server/cni_conf_syncer.go

```diff
@@ -96,18 +98,36 @@ func (syncer *cniNetConfSyncer) syncLoop() error {
 				continue
 			}
 			log.L.Debugf("receiving change event from cni conf dir: %s", event)
-
-			// If the confDir is removed, stop watching.
-			if event.Name == syncer.confDir && (event.Has(fsnotify.Rename) || event.Has(fsnotify.Remove)) {
-				return fmt.Errorf("cni conf dir is removed, stop watching")
-			}
-
 			lerr := syncer.netPlugin.Load(syncer.loadOpts...)
```

Comment:
running load immediately after removing the load directory is going to fail

Reply:
I think we don't need to care about  run and stop cri containers, If /etc/cni/net.d is renamed or 10-calico.conflist  is deleted
kubelet will failed to delete and create pods. 
```
[root@node1 net.d]# ls /etc/cni/net.d
10-calico.conflist  calico-kubeconfig
```
I think we use https://github.com/fsnotify/fsnotify the original purpose is we can update files under dir  /etc/cni/net.d and containerd will auto load.

Reply:
some of the stop operations will work, namely the stop will KILL all the containers in the pod... then fail to call CNI to clean up the network namespace's networks the the configured calls to the CNI plugins.. 

Worse.. if you actually modified the configuration such as by changing the networking used for pods, it is unlikely that the pod network teardown will work ... leaving the admin to restore CNI to original.. acquiesce the node bring down the node.. do the upgrade .. then restart the node